### PR TITLE
Address False Positives for Autosar A13-5-2

### PR DESCRIPTION
**Pull request by jeongsoolee09**
## Description

Address false positives for Autosar A13-5-2.

Slapped `isCompilerGenerated` onto the `where` clause and now lambdas with empty capture is excluded: they were compiled internally to a set of nodes where one of them was for whatever reason a `ConversionOperator`.

## Change request type

 - [ ] Release or process automation (GitHub workflows, internal scripts)
 - [ ] Internal documentation
 - [ ] External documentation
 - [X] Query files (`.ql`, `.qll`, `.qls` or unit tests)
 - [ ] External scripts (analysis report or other code shipped as part of a release)

## Rules with added or modified queries

 - [ ] No rules added
 - [ ] Queries have been added for the following rules:
     - _rule number here_
 - [X] Queries have been modified for the following rules:
     - _rule number here_

## Release change checklist

A change note ([development_handbook.md#change-notes](https://github.com/github/codeql-coding-standards/blob/main/docs/development_handbook.md#change-notes)) is required for any pull request which modifies:

 - The structure or layout of the release artifacts.
 - The evaluation performance (memory, execution time) of an existing query.
 - The results of an existing query in any circumstance.

If you are only adding new rule queries, a change note is not required.

_**Author:**_ Is a change note required? 
  - [ ] Yes
  - [X] No

🚨🚨🚨
_**Reviewer:**_ Confirm that format of *shared* queries (not the .qll file, the
.ql file that imports it) is valid by running them within VS Code. 
  - [ ] Confirmed

_**Reviewer:**_ Confirm that either a change note is not required or the change note is required and has been added.
  - [ ] Confirmed

## Query development review checklist

For PRs that add new queries or modify existing queries, the following checklist should be completed by both the author and reviewer:

### Author

 - [ ] Have all the relevant rule package description files been checked in?
 - [ ] Have you verified that the metadata properties of each new query is set appropriately?
 - [ ] Do all the unit tests contain both "COMPLIANT" and "NON_COMPLIANT" cases?
 - [ ] Are the alert messages properly formatted and consistent with the [style guide](https://github.com/github/codeql-coding-standards/blob/main/docs/development_handbook.md#query-style-guide)?
 - [ ] Have you run the queries on OpenPilot and verified that the performance and results are acceptable?<br />_As a rule of thumb, predicates specific to the query should take no more than 1 minute, and for simple queries be under 10 seconds. If this is not the case, this should be highlighted and agreed in the code review process._
 - [ ] Does the query have an appropriate level of in-query comments/documentation?
 - [ ] Have you considered/identified possible edge cases?
 - [ ] Does the query not reinvent features in the standard library?
 - [ ] Can the query be simplified further (not golfed!)

### Reviewer 

 - [ ] Have all the relevant rule package description files been checked in?
 - [ ] Have you verified that the metadata properties of each new query is set appropriately?
 - [ ] Do all the unit tests contain both "COMPLIANT" and "NON_COMPLIANT" cases?
 - [ ] Are the alert messages properly formatted and consistent with the [style guide](https://github.com/github/codeql-coding-standards/blob/main/docs/development_handbook.md#query-style-guide)?
 - [ ] Have you run the queries on OpenPilot and verified that the performance and results are acceptable?<br />_As a rule of thumb, predicates specific to the query should take no more than 1 minute, and for simple queries be under 10 seconds. If this is not the case, this should be highlighted and agreed in the code review process._
 - [ ] Does the query have an appropriate level of in-query comments/documentation?
 - [ ] Have you considered/identified possible edge cases?
 - [ ] Does the query not reinvent features in the standard library?
 - [ ] Can the query be simplified further (not golfed!)

### CODE REVIEW

File: cpp/autosar/src/rules/A13-5-2/UserDefinedConversionOperatorsNotDefinedExplicit.ql [hunk 1]
```diff
@@ -15,6 +15,7 @@
 import cpp
 import codingstandards.cpp.autosar
 
+// import semmle.code.cpp.PrintAST
```
Comment (lcartey) [inline cpp/autosar/src/rules/A13-5-2/UserDefinedConversionOperatorsNotDefinedExplicit.ql:18]: Remove commented out code.

File: cpp/autosar/src/rules/A13-5-2/UserDefinedConversionOperatorsNotDefinedExplicit.ql [hunk 2]
```diff
@@ -27,5 +28,7 @@ class ExplicitConversionOperator extends ConversionOperator {
 from ConversionOperator op
 where
   not isExcluded(op, OperatorsPackage::userDefinedConversionOperatorsNotDefinedExplicitQuery()) and
-  not op instanceof ExplicitConversionOperator
+  not op instanceof ExplicitConversionOperator and
+  not op.isCompilerGenerated()
 select op, "User-defined conversion operator is not explicit."
+// select 1
```
Comment (lcartey) [inline cpp/autosar/src/rules/A13-5-2/UserDefinedConversionOperatorsNotDefinedExplicit.ql:34]: Remove commented out code.

**Review by lcartey — CHANGES_REQUESTED** [review]
This also needs a change note, otherwise looks good!

File: cpp/autosar/src/rules/A13-5-2/UserDefinedConversionOperatorsNotDefinedExplicit.ql (hunk 2 above)
Comment (jeongsoolee09) [reply cpp/autosar/src/rules/A13-5-2/UserDefinedConversionOperatorsNotDefinedExplicit.ql:34]: 🤦‍♂️

File: change_notes/2022-02-23-a13-5-2-fix-reported-fp-for-a13-5-2.md [hunk 3]
```diff
@@ -0,0 +1 @@
+- Address a false positive on `a13-5-2` where lambda expressions with empty captures were being flagged as having a non-compliant conversion operator.
```
Comment (lcartey) [inline change_notes/2022-02-23-a13-5-2-fix-reported-fp-for-a13-5-2.md:1]:
```suggestion
- `A13-5-2` - address a false positive where lambda expressions with empty captures were being flagged as having a non-compliant conversion operator.
```

**Review by lcartey — APPROVED** [review]
One minor suggested format change, otherwise looks good to me!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **clang/c/x86_64** Matrix Testing for this PR has been completed but I didn't find anything to test!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **gcc/cpp/x86_64** Matrix Testing for this PR has been completed. See below for the results! <br><br>
``` 

TEST_DIFFERENCE      : 
QUERY                : UserDefinedConversionOperatorsNotDefinedExplicit
COMPILE_ERROR_OUTPUT : 
RULE                 : A13-5-2
PACKAGE              : Operators
SUITE                : AUTOSAR
TEST_PASS            : True
COMPILE_PASS         : True


```

**Comment by jsinglet** [comment]
🤖 Beep Boop! **clang/cpp/x86_64** Matrix Testing for this PR has been completed. See below for the results! <br><br>
``` 

COMPILE_ERROR_OUTPUT : 
TEST_PASS            : True
COMPILE_PASS         : True
RULE                 : A13-5-2
PACKAGE              : Operators
TEST_DIFFERENCE      : 
SUITE                : AUTOSAR
QUERY                : UserDefinedConversionOperatorsNotDefinedExplicit


```

**Comment by jsinglet** [comment]
🤖 Beep Boop! Matrix Testing for this PR has been **completed**. If no reports were posted it means this PR does not contain things that need matrix testing!

File: change_notes/2022-02-23-a13-5-2-fix-reported-fp-for-a13-5-2.md (hunk 3 above)
Comment (lcartey) [reply change_notes/2022-02-23-a13-5-2-fix-reported-fp-for-a13-5-2.md:1]: @jeongsoolee09 I think just this item to address, and then we can get this merged.

**Comment by jsinglet** [comment]
🤖 Beep Boop! **gcc/c/x86_64** Matrix Testing for this PR has been completed but I didn't find anything to test!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **gcc/cpp/x86_64** Matrix Testing for this PR has been completed. See below for the results! <br><br>
``` 

RULE                 : A13-5-2
TEST_DIFFERENCE      : 
COMPILE_ERROR_OUTPUT : 
TEST_PASS            : True
SUITE                : AUTOSAR
PACKAGE              : Operators
QUERY                : UserDefinedConversionOperatorsNotDefinedExplicit
COMPILE_PASS         : True


```

**Comment by jsinglet** [comment]
🤖 Beep Boop! **clang/cpp/x86_64** Matrix Testing for this PR has been completed. See below for the results! <br><br>
``` 

QUERY                : UserDefinedConversionOperatorsNotDefinedExplicit
COMPILE_PASS         : True
TEST_PASS            : True
SUITE                : AUTOSAR
RULE                 : A13-5-2
TEST_DIFFERENCE      : 
COMPILE_ERROR_OUTPUT : 
PACKAGE              : Operators


```

**Comment by jsinglet** [comment]
🤖 Beep Boop! Matrix Testing for this PR has been **completed**. If no reports were posted it means this PR does not contain things that need matrix testing!

**Pull request merged**